client/packages/system/src/Location/api/hooks/useLocation.ts

```diff
@@ -89,3 +101,53 @@ const useUpdateLocation = () => {
     },
   });
 };
+
+const useDeleteLocation = (locations?: LocationRowFragment[]) => {
+  const { locationApi, queryClient, storeId } = useLocationGraphQL();
+
+  const { selectedRows } = useTableStore(state => ({
+    selectedRows: Object.keys(state.rowState)
+      .filter(id => state.rowState[id]?.isSelected)
+      .map(selectedId => locations?.find(({ id }) => selectedId === id))
+      .filter(Boolean) as LocationRowFragment[],
+  }));
+
+  const mutationFn = async (id: string) => {
+    const result = await locationApi.deleteLocation({
+      input: {
+        id,
+      },
+      storeId,
+    });
+    return result.deleteLocation;
+  };
+
+  const { mutateAsync: deleteMutation } = useMutation({
+    mutationFn,
+    onSuccess: () => {
+      queryClient.invalidateQueries([LOCATION]);
+    },
+  });
+
+  const deleteLocations = async () => {
+    const deleteErrors: DeleteError[] = [];
+
+    await Promise.all(
+      selectedRows.map(async location => {
+        const data = await deleteMutation(location.id);
+        if (data?.__typename === 'DeleteLocationError') {
+          deleteErrors.push({
+            locationName: location.name,
+            message: data?.error?.description ?? '',
+          });
+        }
+      })
+    );
+    return deleteErrors;
```

Comment:
👀  needed access to the individual item names, so have the errors returned in api for mapping in the footer...